includes/CLI/Plugin_Check_Command.php

```diff
@@ -289,6 +307,35 @@ public function list_checks( $args, $assoc_args ) {
 			)
 		);
 
+		// Filter by stability.
+		if ( ! empty( $options['stability'] ) ) {
+			$all_checks = array_filter(
+				$all_checks,
+				static function ( $item ) use ( $options ) {
+					return $options['stability'] === $item['stability'];
+				}
+			);
+		}
```

Comment:
See above: This shouldn't be needed as we should instead only retrieve experimental checks in the first place if the flag is set.